src/Command/CreateArticle.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Command;
+
+class CreateArticle implements Command
+{
+    /**
+     * @var string
+     */
+    public $title;
+
+    /**
+     * @var string
+     */
+    public $content;
```

Comment:
If you’re not using PHP 7.4’s typed properties, I’d still use private properties and public getters (and setters) with a typed return, in order to guarantee a certain level of integrity to the consumers of this DTO.

Also, as a side note, it’s not the case here, but sometimes you’ll definitely need private properties without a setter and whose value you’d inject through the DTO’s constructor, in order to make them basically read-only. This can be useful to pass some information to the domain that are not supplied by the user. For example, if users must hit a `[PUT] /articles/{id}` endpoint in order to modify an article, you’d want, in your controller, to pass the `id` parameter to the domain, but not allow users to pass it through their request body. I don’t know if you see what I mean?

Reply:
Crystal clear :ok_hand: